rskj-core/src/integrationTest/java/pte/PteIntegrationTest.java

```diff
@@ -0,0 +1,270 @@
+                                accounts.get(4), accounts.get(5), accounts.get(6), accounts.get(7));
+
+                        txResponseMap.put("bulkTransactionsResponse", txResponse);
+
+                        // Await for n blocks to be mined and return them
+
+                        Future<Map<Integer, String>> future = getBlocksAsync();
+
+                        try {
+
+                            blocksResponseMap.put("asyncBlocksResult", future.get());
+
+                        } catch (ExecutionException | InterruptedException e) {
+                            e.printStackTrace();
+                            Assertions.fail(e);
+                        }
+
+                    } catch (IOException e) {
+                        Assertions.fail(e);
+                    }
+                }
+        );
+
+        // Then
+
+        Assertions.assertEquals(200, txResponseMap.get("bulkTransactionsResponse").code());
+
+        Map<Integer, String> blocksResult = blocksResponseMap.get("asyncBlocksResult");
+        Assertions.assertEquals(MAX_BLOCKS_TO_GET, blocksResult.size());
+
+        boolean pteFound = false;
+        int i = blocksResult.size(); // Start from the last element (optimization)
+        while (!pteFound && i >= 0) {
+            i -= 1;
+
+            JsonNode blockResponse = objectMapper.readTree(blocksResult.get(i));
+            JsonNode result = blockResponse.get(0).get("result");
+
+            if (!result.isNull()) {
+                JsonNode pteEdges = result.get("rskPteEdges");
+                if (pteEdges.isArray() && pteEdges.size() > 0) {
+                    Assertions.assertTrue(result.get("transactions").isArray());
+                    Assertions.assertTrue(result.get("transactions").size() > 1);
+                    pteFound = true;
+                }
+            }
+        }
+
+        Assertions.assertTrue(pteFound);
+
+    }
+
+    private Response getBlockByNumber(String number) throws IOException {
+        String content = "[{\n" +
+                "    \"method\": \"eth_getBlockByNumber\",\n" +
+                "    \"params\": [\n" +
+                "        \"" + number + "\",\n" +
+                "        true\n" +
+                "    ],\n" +
+                "    \"id\": 1,\n" +
+                "    \"jsonrpc\": \"2.0\"\n" +
+                "}]";
+
+        return OkHttpClientTestFixture.sendJsonRpcMessage(content, RPC_PORT);
+    }
+
+    private Future<Map<Integer, String>> getBlocksAsync() {
+        CompletableFuture<Map<Integer, String>> completableFuture = new CompletableFuture<>();
+
+        Executors.newCachedThreadPool().submit(() -> {
+            Map<Integer, String> results = new HashMap<>();
+
+            for (int i = 0; i < MAX_BLOCKS_TO_GET; i++) {
+                String response = getBlockByNumber("0x" + String.format("%02x", i)).body().string();
+
+                results.put(i, response);
+                Thread.sleep(500);
```

Comment:
if it works without `sleep()`, I'd rather do not put this delay here

Reply:
The delay here is needed in order to wait for blocks to be processed, I did this on a separate thread in order to avoid blocking the main one.